src/main/groovy/com/vherasymenko/avro/decoder/core/AvroDecoderService.groovy

```diff
@@ -0,0 +1,25 @@
+package com.vherasymenko.avro.decoder.core
+
+import event.course_install.CourseInstall
+import org.apache.avro.generic.GenericDatumReader
+import org.apache.avro.generic.GenericRecord
+import org.apache.avro.io.DatumReader
+import org.apache.avro.io.DecoderFactory
+
+/**
+ * The avro decoder.
+ */
+class AvroDecoderService implements AvroDecoderPort {
+
+    @Override
+    void decodeEvent( String event ) {
+        def inputStream = new ByteArrayInputStream( event.bytes )
+        def schema = CourseInstall.classSchema
+
+        def decoder = new DecoderFactory().jsonDecoder( schema, inputStream )
+        DatumReader<GenericRecord> reader = new GenericDatumReader<>( schema )
+        def document = reader.read( null, decoder )
```

Comment:
I know it's simply an input that we don't care about, but even a variable name would be easier to get across the information. For instance if it was a document position doing:
def documentPosition = null
Will yield a lot more context to the reader. Looking at it now I agree that we cannot pin down what you're setting to null.